internal/blocksync/reactor.go

```diff
@@ -520,12 +549,13 @@ func (r *Reactor) poolRoutine(ctx context.Context, stateSynced bool, blockSyncCh
 
 			// see if there are any blocks to sync
 			first, second, extCommit := r.pool.PeekTwoBlocks()
-			if first == nil || second == nil || extCommit == nil {
-				if first != nil && extCommit == nil {
-					// See https://github.com/tendermint/tendermint/pull/8433#discussion_r866790631
-					panic(fmt.Errorf("peeked first block without extended commit at height %d - possible node store corruption", first.Height))
-				}
-				// we need all to sync the first block
+			if state.ConsensusParams.ABCI.VoteExtensionsEnabled(state.LastBlockHeight+1) &&
```

Comment:
(suggestion) Do you think it would be more readable if we picked the height from `first`?
Something like this:
```go
if first != nil && state.ConsensusParams.ABCI.VoteExtensionsEnabled(first.height+1) &&
     && extCommit == nil {
```

Reply:
BTW, why the "+1" ?

Reply:
LastHeight is the previous height, hence+1, first.Height is the height we're about to process. Either work, but first.Height is fine too, switched.